cachelib/navy/Factory.cpp

```diff
@@ -460,7 +481,11 @@ std::unique_ptr<Device> createFileDevice(
   for (const auto& path : filePaths) {
     folly::File f;
     try {
-      f = openCacheFile(path, fdSize, truncateFile);
+      if(isFDPEnabled) {
+        f = openNvmeCharFile(path);
+      } else {
+        f = openCacheFile(path, fdSize, truncateFile);
+      }
```

Comment:
IIUC, your requirement is as follows,
There are partitions like nvmeXnYp1, nvmeXnYp2 ... , and one of them is used in this cachelib instance.
(Ex: nvmCachePath = "/dev/nvmeXnYp2")
OR is it just one partition in that namespace, but it happened to be partition-formatted ?

>> The rationale is that this will allow us to apply the size check and fallocate thing the same
Is this steps used for device simulation scenarios?

Anyway the proposed approach is doable and making sense.